mesh/ncmesh.cpp

```diff
@@ -4253,6 +4253,19 @@ void NCMesh::GetPointMatrix(Geometry::Type geom, const char* ref_path,
             pm = PointMatrix(mid12, mid20, mid01);
          }
       }
+      else if (geom == Geometry::SEGMENT)
+      {
+         Point mid01(pm(0), pm(1));
+
+         if (child == 0)
+         {
+            pm = PointMatrix(pm(0), mid01);
+         }
+         else if (child == 1)
+         {
+            pm = PointMatrix(mid01, pm(1));
+         }
+      }
```

Comment:
This is clear, Geometry::SEGMENT needs to be here. I would also add a final `else` branch with an `MFEM_ABORT("unsupported geometry " << geom);`, like everywhere else (this should probably have been here all along).